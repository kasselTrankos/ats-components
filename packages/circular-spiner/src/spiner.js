import React, {useState, useEffect} from 'react';
import { View, Animated, Easing, Text} from 'react-native';
import Svg, { Path} from 'react-native-svg';

const AnimatedPath = Animated.createAnimatedComponent(Path);

const polarToCartesian = (centerX, centerY, radius, angleInDegrees) => {
  var angleInRadians = (angleInDegrees-90) * Math.PI / 180.0;

  return {
    x: centerX + (radius * Math.cos(angleInRadians)),
    y: centerY + (radius * Math.sin(angleInRadians))
  };
}
const rnd = arr => arr[Math.floor(Math.random()*arr.length)]
const describeArc = (x, y, radius, startAngle, endAngle) =>{

  var start = polarToCartesian(x, y, radius, endAngle);
  var end = polarToCartesian(x, y, radius, startAngle);

  var largeArcFlag = endAngle - startAngle <= 180 ? "0" : "1";

  var d = [
      "M", start.x, start.y,
      "A", radius, radius, 0, largeArcFlag, 0, end.x, end.y
  ].join(" ");

  return d;
}



const Spiner = props => {
  const {
    dialWidth = 40,
    timeIn = 2000,
    timeOut = 700,
    radius =180,
    opacity = 0.5,
    backgroundColor = '#999',
    colors = ['#505AFC', '#8527B7', '#F2BE22']
  } = props;
  const spinerWheel = (color, index) => {
    const wheel  = new Animated.Value(0);
    const anim = () => {
      Animated.sequence([
        Animated.delay(500 * index),
        Animated.timing(
          wheel,
          {
            toValue: 1,
            duration: timeIn,
            useNativeDriver: true
          }
        ),
        Animated.timing(
          wheel,
          {
            toValue: 0,
            duration: timeOut,
            useNativeDriver: true
          }
        ),
      // Call on self when animation completes
      ]).start(anim);
    };
    const dRange = [];
    const iRange = [];
    const strokeWidth = Math.floor(Math.random() * dialWidth);
    const start = Math.floor(Math.random() * 360);
    const steps = start + 360;
    for (let i = start; i < steps; i++){
     
      dRange.push(describeArc(radius , radius, radius/2, start, 0 + i));
      iRange.push((i - start)/(360 -1));
    }
    const rotation = wheel.interpolate({
      inputRange: iRange,
      outputRange: dRange
    });
    return {jsx: 
      <AnimatedPath 
        key={index} 
        d={rotation}
        stroke={color} 
        strokeWidth={strokeWidth} fill="none"/>, fn: anim};
  }
  const anims = colors.map(spinerWheel);
  useEffect(() => {
    anims.forEach(({fn}) => fn())
  }, []);
  return ( <View style={{
      flex:1,
      alignItems: 'center',
      justifyContent: 'center',
    }}>
      <View
        style={{position: 'absolute', top:0, left: 0, backgroundColor, opacity}}
        width="100%"
        height="100%"
      ></View>
      <Svg
        width={radius * 2}
        height={radius *2}>
          {anims.map(({jsx}) => jsx)}
        </Svg>
      
    </View>
  );
};
export default Spiner;    

  
