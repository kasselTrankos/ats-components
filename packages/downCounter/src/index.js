import React, {Component} from 'react';
import { StyleSheet, View, Text } from 'react-native';
const Props = {};
export class downCounter extends Component<Props> {
  state = { counter: 0};  
  constructor(props) {
    super(props);
  }
  _getCounter () {
    return (<Text  style={{ ...StyleSheet.flatten(styles.text) }}>
      {this.state.counter}
    </Text>);
  }
  render() {
    const {
      value = 15
    } = this.props;

    return (
      <View style={{ ...StyleSheet.flatten(styles.container) }}>
        {this._getCounter()}
      </View>
    );
  }
}
const styles = StyleSheet.create({
  container: {
    position:'relative',
  },
  text: {
    fontSize: 20,
    color: '#fff'
  },
});