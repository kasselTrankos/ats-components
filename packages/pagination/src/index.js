import React, {Component} from 'react';
import { StyleSheet, View } from 'react-native';
import { Button, Text} from 'native-base';
const Props = {};
export class Pagination extends Component<Props> {
  render() {
    const {pages=[], page=1, onClick = ()=>{}} = this.props;
    return (
      <View style={styles.pagination}>
          {pages.map(item => 
            <Button
              key={'-'+ item + '-'}
              onPress = {() => onClick(current)}
              style={styles.button}
              info={page!==item} 
              small
              id={item+'-page'}>
              <Text style={styles.paginationText}>{item}</Text>
            </Button>
          )}
        </View>
    );
  }
}
const styles = StyleSheet.create({
  pagination: {
    justifyContent: 'center',
    alignItems: 'center',
    flexDirection: 'row'
  },
  button: {
    marginRight: 2,
    paddingLeft: 8,
    paddingRight: 8
  },
  paginationText: {
    fontFamily: "PT_Sans-Narrow-Web-Regular",
    fontSize: 22,
    color:'#fff',
    paddingLeft: 6,
    paddingRight: 6,
  }
});