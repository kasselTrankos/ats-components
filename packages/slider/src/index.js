
import React, { Component } from 'react';
import {PanResponder, View} from 'react-native';
import {polarToCartesian, cartesianToPolar, getOnPressAngle} from './../lib/math';
import { Svg, G, Defs, Stop, LinearGradient, Circle, Path } from 'react-native-svg';
const CLOCKWISE = "CW";
const COUNTER_CLOCKWISE = "CCW";

export default class CircleSlider extends Component {
  constructor(props) {
    super(props);
    this.handleMeasure = (ox, oy, width, height, px, py) => {
      const center = (this.props.sliderWidth + (this.props.sliderRadius * 2)) / 2;
      this.setState({
        xCenter: px + (this.props.sliderRadius + this.props.btnRadius),
        yCenter: py + (this.props.sliderRadius + this.props.btnRadius),
        measuredBox: this.getBoxBounds(),
        circleCenter: { x: center, y: center }
      }, () => this.props.onValueChange 
        && this.props.onValueChange(this.props.startDegree ? this.props.startDegree : 0));
    };
    this.measureLocation = () => {
      // @ts-ignore
      this.refs.circleslider.measure(this.handleMeasure);
    };
    this.getOnPressAngle = (x, y) => {
      const xOrigin = this.state.xCenter - (this.props.sliderRadius + this.props.btnRadius);
      const yOrigin = this.state.yCenter - (this.props.sliderRadius + this.props.btnRadius);
      const a = (cartesianToPolar(x - xOrigin, y - yOrigin)(this.props));
      const relativeAngle = this.getRelativeAngle(a);
      this.setState({ origin: { x: xOrigin, y: yOrigin } });

      return { angle: a, relativeAngle };
    };
    this.getRelativeAngle = (angle) => {
      const start = this.props.startDegree !== undefined ? this.props.startDegree : 0;
      if (angle < start) {
        return this.props.arcDirection === CLOCKWISE ? (Math.abs(360 - start) + angle) % 360 : start - angle;
      }
      return this.props.arcDirection === CLOCKWISE ? (angle - start) : (start + (360 - angle)) % 360;
    };
    this.relativeToAbsoluteAngle = (relativeAngle) => {
      const start = this.props.startDegree !== undefined ? this.props.startDegree : 0;
      const relative = (this.props.arcDirection === CLOCKWISE ? 1 : -1) * relativeAngle;
      return start + (relative) % 360;
    };
    this.getCurrentValue = () => {
      return parseFloat((this.state.relativeAngle / 360 * this.props.maxValue).toFixed(this.props.decimalPlaces));
    };
    this.setValue = (value) => {
      const rel = ((value * 360) / this.props.maxValue) % 360;
      const a = this.relativeToAbsoluteAngle(rel) % 360;
      this.setState({ angle: a, relativeAngle: rel });
    };
    this.onValueChanged = (value) => {
      if (this.props.onValueChange !== undefined) {
        this.props.onValueChange(value);
      }
    };
    this.innerCirclePressed = (value) => {
      if (this.props.onPressInnerCircle !== undefined) {
        this.props.onPressInnerCircle(value);
      }
    };
    this.outerCirclePressed = (value) => {
      if (this.props.onPressOuterCircle !== undefined) {
        this.props.onPressOuterCircle(value);
      }
    };
    this.getBoxBounds = () => {
      let degree = 45;
      if (this.props.startDegree) {
        degree += this.props.startDegree;
      }
      const arr = [];
      for (var i = 0; i < 4; i++) {
        arr.push((degree + (90 * i)) % 360);
      }
      const pointArray = [];
      arr.forEach((angle) => {
        let x = (this.props.sliderRadius - (this.props.sliderWidth / 2)) * Math.cos(this.degreeToRadian(angle)) + this.state.circleCenter.x;
        let y = (this.props.sliderRadius - (this.props.sliderWidth / 2)) * Math.sin(this.degreeToRadian(angle)) + this.state.circleCenter.y;
        let newPoint = {
          angle: this.relativeToAbsoluteAngle(angle),
          x: x,
          y: y
        };
        pointArray.push(newPoint);
      });
      return pointArray;
    };
    this.degreeToRadian = (angle) => {
      return angle * 0.0174533;
    };
    this.radianToDegree = (radian) => {
      return radian * 57.2958;
    };
    this.state = {
      angle: this.relativeToAbsoluteAngle(((this.props.value !== undefined ? this.props.value : 0) * 360) / this.props.maxValue),
      relativeAngle: ((this.props.value !== undefined ? this.props.value : 0) * 360) / this.props.maxValue,
      xCenter: Number.NEGATIVE_INFINITY,
      yCenter: Number.NEGATIVE_INFINITY,
      origin: { x: 0, y: 0 },
      circleCenter: { x: 0, y: 0 },
      measuredBox: []
    };
    if (props.arcDirection !== CLOCKWISE && props.arcDirection !== COUNTER_CLOCKWISE) {
      throw new Error("Prop 'arcDirection' only supports 'CW' or 'CCW', for Clockwise or Counterclockwise");
    }
    this.panResponder = PanResponder.create({
      onMoveShouldSetPanResponder: (e, gs) => true,
      onMoveShouldSetPanResponderCapture: (e, gs) => true,
      onPanResponderMove: (e, gs) => {
        const angles = this.getOnPressAngle(gs.moveX, gs.moveY);
        this.setState({ angle: angles.angle, relativeAngle: angles.relativeAngle }, () => {
          this.onValueChanged(this.getCurrentValue());
        });
      },
      onStartShouldSetPanResponder: (e, gs) => true,
      onStartShouldSetPanResponderCapture: (e, gs) => true,
    });
  }
  UNSAFE_componentWillReceiveProps(nextProps) {
    if (this.props.value !== nextProps.value) {
      this.setValue((nextProps.value !== undefined ? nextProps.value : 0));
    }
  }
  render() {
    const width = this.props.sliderWidth + (this.props.sliderRadius * 2);
    const bR = this.props.btnRadius;
    const dR = this.props.sliderRadius;
    const angle = this.props.startDegree !== undefined ? this.props.startDegree : 0
    const startCoord = polarToCartesian(angle)(this.props);
    const endCoord = polarToCartesian(this.state.angle)(this.props);

    const radiusX = dR;
    const radiusY = dR;
    const xAxisRotation = 0; // The x-axis-rotation does not have an effect on a circle, 0 is a good choice.
    const largeArc = this.state.relativeAngle >= 180 ? 1 : 0;
    const sweepFlag = this.props.arcDirection === CLOCKWISE ? 1 : 0;
    return (<View>
      <Svg
        style={{width: '100%'}}
        onLayout={this.measureLocation} 
        ref="circleslider" 
        width={width} 
        height={width} 
        flex={1}>
        <Defs>
          <LinearGradient id="gradient1" x1="0%" y1="0%" x2="100%" y2="0%">
            <Stop offset="0%" stopColor={this.props.startGradient} />
            <Stop offset="100%" stopColor={this.props.endGradient} />
          </LinearGradient>
        </Defs>
        <Circle 
          r={dR} 
          cx={width / 2} 
          cy={width / 2} 
          stroke={this.props.backgroundColor} 
          strokeWidth={this.props.sliderWidth} fill="none" />
        <Path 
          stroke={"url(#gradient1)"} 
          strokeWidth={this.props.sliderWidth} 
          fill="none" 
          strokeLinecap="round" 
          strokeLinejoin="round" 
          d={`M${startCoord.x} ${startCoord.y} A ${radiusX} ${radiusY} ${xAxisRotation} ${largeArc} ${sweepFlag} ${endCoord.x} ${endCoord.y}`} 
          onPressIn={(e) => {
            const {locationX, location} = e.nativeEvent;
            const angles = this.getOnPressAngle(locationX, locationY);
            this.setState({ angle: angles.angle, relativeAngle: angles.relativeAngle }, () => {
              const currentValue = this.getCurrentValue();
              this.outerCirclePressed(currentValue);
              this.onValueChanged(currentValue);
            });
          }} />

        <Circle r={dR + ((dR * 25) / 100)} cx={width / 2} cy={width / 2} stroke="none" fill="none" onPressIn={(e) => {
          const {pageX, pageY} = e.nativeEvent;
          const angles = this.getOnPressAngle(pageX, pageY);
          this.setState({ angle: angles.angle, relativeAngle: angles.relativeAngle }, () => {
            const currentValue = this.getCurrentValue();
            this.outerCirclePressed(currentValue);
            this.onValueChanged(currentValue);
          });
        }} />

        <G x={endCoord.x - bR} y={endCoord.y - bR}>
          <Circle r={bR} cx={bR} cy={bR} fill={this.props.btnColor} {...this.panResponder.panHandlers} />
        </G>
        { this.state.measuredBox.length > 0 && <View
          style={{
            overflow: "visible", 
            position: "absolute", 
            top: this.state.measuredBox[2].y, 
            left: this.state.measuredBox[2].x, 
            width: this.state.measuredBox[0].x - this.state.measuredBox[2].x, 
            height: this.state.measuredBox[1].y - this.state.measuredBox[2].y
          }}>{this.props.component}</View>}
      </Svg>
    </View>);
  }
}
CircleSlider.defaultProps = {
  arcDirection: CLOCKWISE,
  backgroundColor: "white",
  btnColor: 'yellow',
  btnRadius: 13,
  component: undefined,
  decimalPlaces: 0,
  endGradient: "#A6FFCB",
  flex: 1,
  maxValue: 360,
  onPressInnerCircle: (value) => value,
  onPressOuterCircle: (value) => value,
  onValueChange: (value) => value,
  showValue: true,
  sliderRadius: 130,
  sliderWidth: 25,
  startDegree: 0,
  startGradient: "#12D8FA",
  textColor: "white",
  textSize: 50,
  value: 0,
};