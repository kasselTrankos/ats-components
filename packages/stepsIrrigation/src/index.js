import React, {Component} from 'react';
import { StyleSheet, View } from 'react-native';
import { Text, Content } from 'native-base';

const Props = {};
export class StepsIrrigation extends Component<Props> {
constructor(props) {
  super(props);
}

render() {
  const { 
    steps = [],
    style = {},
    color = '#fff',
    fontSize = 18,
    text = '---' } = this.props;
  return (
    <View style={style}>
      {steps.map((step) =>
        <Content contentContainerStyle={styles.step} 
          key={`${text}-${step}_iriirg`}>
          <Text style={{ ...StyleSheet.flatten(styles.stepText), ...StyleSheet.flatten(styles.text), color, fontSize: parseInt(fontSize) }}>{text}</Text>
          <Text style={{ ...StyleSheet.flatten(styles.text), color, fontSize: parseInt(fontSize)}}>{step}</Text>
        </Content>
      )}
    </View>);
  }
}
const styles = StyleSheet.create({
  step: {
    flexDirection: 'row',
    paddingBottom: 15
  },
  stepText: {
    paddingRight: 5,
    paddingLeft: 5
  },
  text: {
    color: '#fff',
    fontSize: 20
  }
});
