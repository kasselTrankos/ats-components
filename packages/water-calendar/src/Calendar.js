// @flow

import React, { useState, useRef } from 'react';
  import { View, Dimensions, PanResponder, ScrollView, Vibration } from 'react-native';
import Day from './Day';


const findCell = (rows, radius) => (x, y)=> {
  const right = Math.floor(x / radius);
  const bottom = Math.floor(y / radius);
  return right + rows * bottom;;
};
const lt = (dateA = new Date()) => dateB => +dateA > +dateB;
const getDay = daysInMonth => {
  let month = 0;
  const currentDay = new Date().getDate();
  const currentMonth = new Date().getMonth();
  const isPassedDate = lt();
  return value => {
    const lessDays = daysInMonth.reduce((acc, current, index)=> {
      if(index < month) {
        acc+=current;
      }
      return acc;
    } , 0);
    const day = value - lessDays;
    const current = new Date(new Date().getFullYear(), month, day);
    if(day===daysInMonth[month]) month ++;

    return {
      month: month,
      day: day,
      isToday: currentMonth === month && day === currentDay,
      isPassed: isPassedDate(current)
    }; 
  }
}

const getDaysMonth = () => {
  const daysInMonth = (month, year) => new Date(year, month, 0).getDate();
  return Array.from({length: 12}).map((el, index) => daysInMonth(index+1, new Date().getFullYear()))
};


const Calendar = props => {
  const {
    amount = 200,
    rows = 7,
    colorDayText = '#192965',
    inactiveColor = '#fff',
    activeColor = '#2988B1',
    vibrationDuration = 100,
    passedDay = '#0f4c75',
    currentDay= '#edf7fa',
    onDates=()=>{},
    
  } = props;
  const getDate = getDay(getDaysMonth())
  const view = useRef();
  const [cellStart, setCellStart] = useState(0);
  const [height, setHeight] = useState(300);
  const [top, setTop] = useState(0);
  const [scrollTop, setScrollTop] = useState(0);
  const [dragging, setDragging] = useState(true);
  const {width} = Dimensions.get('window');
  const radius = Math.round((width) / rows);
  const [days, setDays] = useState( Array(amount).fill().map((v, index) => {
    const {day, isToday, isPassed} = getDate(index + 1);
    return {
      selected: false, 
      key: index,
      day, isPassed, isToday
    };
  }));
  const onDatesSelected = () => {
    const selected = days.filter(({selected}) => selected);
    onDates([...selected]);
  };
  const onPress = key => {
    days[key].selected = !days[key].selected;
    setDays([...days]);
    onDatesSelected();
  }
  const activateDays = (start, end) => days.map(({selected, key, day, isPassed, isToday}) => ({
    selected: key >=start && key<=end, 
    key, day, isPassed, isToday
  }))
  const onPresent = evt => view.current.measure((x, y, width, height, pageX, pageY) =>{
    setTop(pageY);
    setHeight(height);
  });
  
  const inside = (x, y) => x >= top && y <= height +top;
  const handleScroll = e => {
    setScrollTop(Number(e.nativeEvent.contentOffset.y));
  }
  const handleMultiple = ()=> {
    Vibration.vibrate(vibrationDuration);
    setDragging(false)
  };
  const getCell = findCell(rows, radius);
  const panResponde = PanResponder.create({
    // prevent children interactuact prevented.
    onMoveShouldSetPanResponderCapture: ({nativeEvent: {pageX, pageY}}) => inside(pageX, pageY) && !dragging,
    onPanResponderGrant: ({nativeEvent: {pageX = 0, pageY = 0}}) => {
      const _top = Number(scrollTop.toFixed(0)) + Number(pageY.toFixed(0)) - Number(top);
      setCellStart(getCell(pageX.toFixed(0), _top));
      setDays([...activateDays(cellStart, cellStart)])
    },
    onPanResponderMove: ({nativeEvent: {pageX = 0, pageY = 0}}) => {
        const _top = Number(scrollTop.toFixed(0)) + Number(pageY.toFixed(0)) - Number(top);
        const cellEnd =  getCell(pageX.toFixed(0), _top);
        const start = Math.min(cellStart, cellEnd);
        const end = Math.max(cellStart, cellEnd);
        setDays([...activateDays(start, end)]);
    },
    onPanResponderRelease: () => {
      setDragging(true);
      onDatesSelected();
    },
    onPanResponderTerminationRequest: () => true,
  });
  
  return (<ScrollView
    onScroll={handleScroll}
    style={{
      position: 'absolute',
      height: 300,
      width: '100%',
    }}>
    <View style={{
      flexDirection: 'row',
      flexWrap: 'wrap',
      alignItems: 'flex-start',
      top: 0,
      width: '100%',
      height: '100%',
    }}
    ref={view}
    onLayout={onPresent}
    {...panResponde.panHandlers}>
    {days.map(({selected, key, day, isToday, isPassed}, index)=> <Day 
      onLongPress={handleMultiple}
      onPress={onPress}
      selected={selected}
      fillColor={ selected ? activeColor : inactiveColor}
      text={day}
      passedDay={passedDay}
      isToday={isToday}
      isPassed={isPassed}
      currentDay={currentDay}
      dataId={key}
      key={key}
      colorDayText={colorDayText}
      radius={radius} />)}
  </View></ScrollView>);
}

export default Calendar;

