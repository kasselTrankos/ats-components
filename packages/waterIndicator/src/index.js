import React, {Component} from 'react';
import { StyleSheet, View } from 'react-native';
import { StepsIrrigation} from '@ats-components/steps-irrigation';
const Props = {};
export class WaterIndicator extends Component<Props> {
  constructor(props) {
    super(props);
  }
  render() {
    const {
      volumen = 100,
      steps = ['3', '2.5', '2', '1.5', '1', '0.5', '0'],
      fontColor = '#FFF', 
      fontSize= 18,
      bgColor = '#00C3FF',
    } = this.props;

    return (
      <View style={styles.container}>
        <View style={{ ...StyleSheet.flatten(styles.cube), fontColor, fontSize }}>
          <View style={{ ...StyleSheet.flatten(styles.traker), bgColor, height: `${volumen}%`}} />
          <StepsIrrigation 
            style={styles.steps} 
            steps={steps}/>
        </View>
      </View>
    );
  }
}
const styles = StyleSheet.create({
  container: {
    position:'relative',
    flex: 1,
    width: '100%',
    marginTop: 0,
    marginBottom:0,
    height: '100%',
  },
  cube: {
    position: 'relative',
    left: '5%',
    width: '90%',
    height:'100%',
    fontSize: 18,
    textAlign: 'center',
  }, 
  traker: {
    position: 'absolute',
    borderBottomLeftRadius: 12,
    borderBottomRightRadius: 12,
    backgroundColor: '#00C3FF',
    height: '100%',
    zIndex: 5,
    left: 0,
    bottom:0,
    width: '100%',
    overflow: 'hidden'
  },
  steps: {
    left: 0,
    zIndex: 9,
    top: '5%',
    width: '70%',
    height: '90%',
  },
  step: {
    flexDirection: 'row',
  },
  stepText: {
    color :'#fff',
    paddingRight: 5,
    paddingLeft: 5
  },
  text: {
    color: '#fff'
  }
});