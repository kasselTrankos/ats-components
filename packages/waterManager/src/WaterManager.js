import React, {useState} from 'react';
import { View, Text, TouchableHighlight } from 'react-native';
import Water from './Water';
import Slider from './Slider';



const WaterManager  = props => {
  const {
    dialWidth = 8,
    dialRadius = 32,
    dialColor = "#5BC0EB",
    dialTextSize = 22,
    dialTextColor = '#fff',
    fillColor = 'none',
    fontSize = 100,
    fontColor = '#3F88C5',
    maxDial = 360,
    minDial = 0,
    meterColor = '#333',
    onChange = (e)=> {}, 
    onPress = () => {console.log('is presssed')}, 
    radius = 140,
    sizes = [1, 2, 3, 'Empty'],
    strokeWidth = 10,
    strokeColor = '#3F88C5',
    value = 120,
    volumen = 100,
    waterColor= '#A9CEF4',
  } = props;
  const width = radius * 2;
  const height = radius * 2;
  return <TouchableHighlight 
    onPress={onPress}
    underlayColor="white"
    style={{
      position: 'absolute', 
      top: 0, 
      flex: 1,
      width: width + 90,
      height: height + dialRadius * 2,
      alignItems: 'center',
      justifyContent: 'center',
    }}>
    <View style={{ width, height: (height + dialRadius * 2), top: 0,}}>
      <View style={{
          top: dialRadius,
          left: 0,
          position: 'absolute',
          // width: width,
          // height: width,
        }}>
        <Water
          radius={radius}
          onPress={onPress}
          volumen={100 - volumen} 
          backgroundColor={waterColor} />
      </View>
      <View>
        <Text style={{
          position: 'absolute',
          fontSize, 
          lineHeight: radius * 2 + dialRadius * 2,
          textAlignVertical: 'center',
          width: '100%', 
          top: 0, 
          textAlign: 'center', 
          color: fontColor,
          }}>{`${volumen}%`}</Text>
      </View>
      <View style={{left: -dialRadius, top: 0}}>
        <Slider 
          dialColor={dialColor}
          dialRadius={dialRadius}
          dialTextSize={dialTextSize}
          dialTextColor={dialTextColor}
          dialWidth={dialWidth}
          fillColor = {fillColor}
          maxDial={maxDial}
          meterColor={meterColor}
          minDial={minDial}
          onChange={onChange}
          strokeWidth={strokeWidth} 
          strokeColor={strokeColor} 
          radius = {radius}
          value={value} />
      </View>
    </View>

  </TouchableHighlight>
};


export default WaterManager;