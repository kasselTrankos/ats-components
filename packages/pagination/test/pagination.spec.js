import 'react-native';
import React from 'react';
import {create, act} from 'react-test-renderer';
import {Pagination} from './../src';
describe('<Pagination>', () => {
  it('renders the correct text', () => {
    let root;
    act(()=> {root = create(<Pagination pages={[1,2,35, 55]}/>);});
    expect(root.toJSON()).toMatchSnapshot();
  });

//   it('validate given props', ()=> {
//     const prop = [1, 4, 9];
//     const text = '???';
//     const ins = create(<Pagination steps={prop} text={text}/>);
//     const steps = ins.root.findByType(Pagination);
//     expect(steps.props.steps).toEqual(prop);
//     expect(steps.props.text).toEqual(text);
//   });
});