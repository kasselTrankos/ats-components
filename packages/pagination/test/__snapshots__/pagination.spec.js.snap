// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Pagination> renders the correct text 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "flexDirection": "row",
      "justifyContent": "center",
    }
  }
>
  <View
    accessible={true}
    focusable={true}
    isTVSelectable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "#007aff",
        "borderBottomWidth": null,
        "borderColor": "#007aff",
        "borderLeftWidth": null,
        "borderRadius": 5,
        "borderRightWidth": null,
        "borderTopWidth": null,
        "borderWidth": undefined,
        "elevation": 2,
        "flexDirection": "row",
        "height": 30,
        "justifyContent": "space-between",
        "marginRight": 2,
        "opacity": 1,
        "paddingBottom": 6,
        "paddingLeft": 8,
        "paddingRight": 8,
        "paddingTop": 6,
        "shadowColor": undefined,
        "shadowOffset": undefined,
        "shadowOpacity": undefined,
        "shadowRadius": undefined,
      }
    }
  >
    <Text
      style={
        Array [
          Object {
            "backgroundColor": "transparent",
            "color": "#fff",
            "fontFamily": "System",
            "fontSize": 14,
            "marginLeft": 0,
            "marginRight": 0,
            "paddingLeft": 16,
            "paddingRight": 16,
          },
          Object {
            "color": "#fff",
            "fontFamily": "PT_Sans-Narrow-Web-Regular",
            "fontSize": 22,
            "paddingLeft": 6,
            "paddingRight": 6,
          },
        ]
      }
      uppercase={false}
    >
      1
    </Text>
  </View>
  <View
    accessible={true}
    focusable={true}
    isTVSelectable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "#62B1F6",
        "borderBottomWidth": null,
        "borderColor": "#007aff",
        "borderLeftWidth": null,
        "borderRadius": 5,
        "borderRightWidth": null,
        "borderTopWidth": null,
        "borderWidth": undefined,
        "elevation": 2,
        "flexDirection": "row",
        "height": 30,
        "justifyContent": "space-between",
        "marginRight": 2,
        "opacity": 1,
        "paddingBottom": 6,
        "paddingLeft": 8,
        "paddingRight": 8,
        "paddingTop": 6,
        "shadowColor": undefined,
        "shadowOffset": undefined,
        "shadowOpacity": undefined,
        "shadowRadius": undefined,
      }
    }
  >
    <Text
      style={
        Array [
          Object {
            "backgroundColor": "transparent",
            "color": "#fff",
            "fontFamily": "System",
            "fontSize": 14,
            "marginLeft": 0,
            "marginRight": 0,
            "paddingLeft": 16,
            "paddingRight": 16,
          },
          Object {
            "color": "#fff",
            "fontFamily": "PT_Sans-Narrow-Web-Regular",
            "fontSize": 22,
            "paddingLeft": 6,
            "paddingRight": 6,
          },
        ]
      }
      uppercase={false}
    >
      2
    </Text>
  </View>
  <View
    accessible={true}
    focusable={true}
    isTVSelectable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "#62B1F6",
        "borderBottomWidth": null,
        "borderColor": "#007aff",
        "borderLeftWidth": null,
        "borderRadius": 5,
        "borderRightWidth": null,
        "borderTopWidth": null,
        "borderWidth": undefined,
        "elevation": 2,
        "flexDirection": "row",
        "height": 30,
        "justifyContent": "space-between",
        "marginRight": 2,
        "opacity": 1,
        "paddingBottom": 6,
        "paddingLeft": 8,
        "paddingRight": 8,
        "paddingTop": 6,
        "shadowColor": undefined,
        "shadowOffset": undefined,
        "shadowOpacity": undefined,
        "shadowRadius": undefined,
      }
    }
  >
    <Text
      style={
        Array [
          Object {
            "backgroundColor": "transparent",
            "color": "#fff",
            "fontFamily": "System",
            "fontSize": 14,
            "marginLeft": 0,
            "marginRight": 0,
            "paddingLeft": 16,
            "paddingRight": 16,
          },
          Object {
            "color": "#fff",
            "fontFamily": "PT_Sans-Narrow-Web-Regular",
            "fontSize": 22,
            "paddingLeft": 6,
            "paddingRight": 6,
          },
        ]
      }
      uppercase={false}
    >
      35
    </Text>
  </View>
  <View
    accessible={true}
    focusable={true}
    isTVSelectable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "#62B1F6",
        "borderBottomWidth": null,
        "borderColor": "#007aff",
        "borderLeftWidth": null,
        "borderRadius": 5,
        "borderRightWidth": null,
        "borderTopWidth": null,
        "borderWidth": undefined,
        "elevation": 2,
        "flexDirection": "row",
        "height": 30,
        "justifyContent": "space-between",
        "marginRight": 2,
        "opacity": 1,
        "paddingBottom": 6,
        "paddingLeft": 8,
        "paddingRight": 8,
        "paddingTop": 6,
        "shadowColor": undefined,
        "shadowOffset": undefined,
        "shadowOpacity": undefined,
        "shadowRadius": undefined,
      }
    }
  >
    <Text
      style={
        Array [
          Object {
            "backgroundColor": "transparent",
            "color": "#fff",
            "fontFamily": "System",
            "fontSize": 14,
            "marginLeft": 0,
            "marginRight": 0,
            "paddingLeft": 16,
            "paddingRight": 16,
          },
          Object {
            "color": "#fff",
            "fontFamily": "PT_Sans-Narrow-Web-Regular",
            "fontSize": 22,
            "paddingLeft": 6,
            "paddingRight": 6,
          },
        ]
      }
      uppercase={false}
    >
      55
    </Text>
  </View>
</View>
`;
